airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1175,3 +1177,50 @@ async def test_streams_define_primary_key(
 
         quoted_missing_primary_keys = {f"'{primary_key}'" for primary_key in missing_primary_keys}
         assert not missing_primary_keys, f"The following streams {', '.join(quoted_missing_primary_keys)} do not define a primary_key"
+
+    async def test_certified_connector_has_allowed_hosts(self, operational_certification_test, connector_metadata: dict) -> None:
+        """
+        Checks whether or not the connector has `allowedHosts` and it's components defined in `metadata.yaml`.
+        Suitable for certified connectors starting `ql` >= 400.
+
+        Arguments:
+            :: operational_certification_test -- pytest.fixure defines the connector is suitable for this test or not.
+            :: connector_metadata -- `metadata.yaml` file content
+        """
+        metadata = connector_metadata.get("data", {})
+
+        has_allowed_hosts_property = "allowedHosts" in metadata.keys()
+        assert has_allowed_hosts_property, f"The `AllowedHosts` property is missing in `metadata.data` for `metadata.yaml`."
+
+        allowed_hosts = metadata.get("allowedHosts", {})
+        has_hosts_property = "hosts" in allowed_hosts.keys() if allowed_hosts else False
+        assert has_hosts_property, f"The `hosts` property is missing in `metadata.data.AllowedHosts` for `metadata.yaml`."
+
+        hosts = allowed_hosts.get("hosts", [])
```

Comment:
I assume this part for each test could be questionable, because there could be cases like:
```yaml
ql: 400
allowedHosts:
   hosts: []
suggestedStreams:
  streams: []
```
The possible solution is  `hosts:  ["*"]` meaning all hosts are allowed

Reply:
Added `bypass_reason` checks to overcome the situations like described above.

Reply:
i'm okay with a bypass reason, for the moment I think we can hold off on defining exactly what the all hosts are allowed shape looks like because I don't think we've yet defined what that would look like on the platform side.